### PR TITLE
Ensure that user provided SCSS has precedence over quarto generated scss  also for Dark theme

### LINKED ISSUE #10817: Dark theme doesn't respect the layering order of the stylesheets
### Bug description

`boostrap-dark.min.css` doesn't follow the layering order specified in the [documentation](https://quarto.org/docs/output-formats/html-themes-more.html). At the end of the css file, you can find quarto related styles such as `quarto-title-banner`, `quarto-title`, `#title-block-header`, etc which should come before the user-specified styles, not after. Note that this is only an issue with the Dark theme.

### Steps to reproduce

**_quarto.yml**
```
project:  
    type: website

website:
  title: "test2"
  navbar:
    left:
      - href: index.qmd
        text: Home
      - about.qmd

format:
  html:
    theme:
      light:
        - cosmo
        - custom.css
      dark: 
        - darkly
        - custom.css
    toc: true
```

**custom.css**
```
/*-- scss:rules --*/

.custom-style {
    color: red;
}
```

### Expected behavior

**boostrap-dark.min.css**
```
[ bootstrap styles ]

...

.custom-style {
    color: red;
}
```

### Actual behavior

**boostrap-dark.min.css**
```
[ bootstrap styles  - part 1 ]

...

.custom-style {
    color: red;
}

[ bootstrap styles  - part 2 ]

.quarto-title-banner {
    margin-bottom: 1em;
    color: #dee2e6;
    background: #375a7f
}

.quarto-title-banner a {
    color: #dee2e6
}

...

.quarto-title-tools-only {
    display: flex;
    justify-content: right
}
```

### Your environment

- IDE: VSCode 1.92.2
- OS: macOS Sonoma 14.3

### Quarto check output

```
Quarto 1.5.55
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.55
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)
```

### PR DESCRIPTION
**Pull request by cderv**
This is a follow up on https://github.com/quarto-dev/quarto-cli/commit/b7a5449b210302a96952cadb5469dbf30e6db375 which fixed that for the light theme

We do apply the same fix for dark theme. 

fixes https://github.com/quarto-dev/quarto-cli/issues/10817

Now that playwright test are easier to write, this adds a test to check the CSS value defined is the correct one from user's custom theme file

### CODE REVIEW

**Comment by cderv** [comment]
Failure in playwright test don't make the test fails and CI erroring. So I need to improve that... 🤔

File: .github/workflows/test-smokes.yml [hunk 1]
```diff
@@ -283,8 +283,17 @@ jobs:
           echo "Pull Request URL - ${{ steps.cpr.outputs.pull-request-url }}"
           echo "Pull Request Action Performed - ${{ steps.cpr.outputs.pull-request-operation }}"
 
-      - uses: actions/upload-artifact@v3
+      - name: Upload test timing file
```
Comment (cscheid) [inline .github/workflows/test-smokes.yml:286]: Does this belong on his PR?

**Review by cscheid — COMMENTED** [review]
Looks good aside from maybe a stray extra file.

File: .github/workflows/test-smokes.yml (hunk 1 above)
Comment (cderv) [reply .github/workflows/test-smokes.yml:286]: It is just setting the name on the workflow step. I could have done it outside, just did it there while adding an upload of the playwright report.

**Pull request merged**